src/main/java/com/google/devtools/build/lib/query2/cquery/KnownTargetsDependencyResolver.java

```diff
@@ -0,0 +1,35 @@
+package com.google.devtools.build.lib.query2.cquery;
+
+import com.google.devtools.build.lib.analysis.DependencyKind;
+import com.google.devtools.build.lib.analysis.DependencyResolver;
+import com.google.devtools.build.lib.analysis.TargetAndConfiguration;
+import com.google.devtools.build.lib.causes.Cause;
+import com.google.devtools.build.lib.cmdline.Label;
+import com.google.devtools.build.lib.collect.nestedset.NestedSetBuilder;
+import com.google.devtools.build.lib.packages.Target;
+import com.google.devtools.build.lib.util.OrderedSetMultimap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class KnownTargetsDependencyResolver extends DependencyResolver {
```

Comment:
This is exactly a `FormatterDependencyResolver` - no new filter, it was just extracted to be a top-level class. Because `FormatterDependencyResolver` uses state from its outer class, extracting it out to be explicit about which state it depends on felt useful. The problem is that we need to use it from `ProtoOutputFormatterCallback` as well as `TransitionsOutputFormatterCallback` - these are separately invoked by different options to `cquery`, so in order to use this logic from both it either needs extracting or duplicating.